src/api-transition-helpers.coffee

```diff
@@ -0,0 +1,17 @@
+LineNumberGutterName = 'line-numbers'
+
+# Given decorationParams/Properties from the 1.0 API, applies changes to enable
+# the 2.0 API under the covers. In 2.0, the `trueType` property will become `type`.
+translateDecorationParamsOldToNew = (decorationParams) ->
```

Comment:
Just to step back and lay out the tradeoffs:

== Option 1 ==
One of the original goals was to use the new gutter API in Atom Core. We figured this could work if we translated the inputs to ::decorateMarker and returned objects that conformed to the new API. This would look like:
* type: `line-number` --> type: `gutter`, name: `line-number`
* type: `gutter` (no name specified) --> same as above
* type: `gutter`, name: `custom-name` --> no change
This would break the API for people getting Decorations though, because they would expect a `line-number` gutter type on the `line-number` gutter.

Furthermore, with this translation, there would be confusion when getting decorations. If type: `gutter` is specified, there would be confusion about whether it should include the custom decorations. Specifically, what would be expected to happen is that when `::getDecorations` is called externally (from outside Atom Core), then we don't return any custom decorations. But Atom Core should be conforming to the new API, and thus should expect all `gutter` type decorations, including ones on custom gutters.


== Option 2 ==
We can let go of the goal to keep Atom Core using the new gutter API. 
Then we can do the following translations when setting decorations (which does not break the API for external users):
* type: `line-number` --> type: `line-number`, name: `line-number`
* type: `gutter` (no name specified) --> same as above
* type: `gutter`, name: `custom-name` --> no change

Then, when getting the line-number decorations, in Atom Core code, we would have to use type='line-gutter' instead of type='gutter' && name='line-number'. So internally, we wouldn't be using the new API, and these spots would have to be identified and changed later. 


== Option 3 ==
This is the route I went, which was to try to allow for an API that maintains the public API, and allows us to call things in the "new" way internally. I think there isn't a way to get around storing more state, which is the downside to this option. 


I can go with any of these. It sounds like you are advocating for Option 2?